sql/core/src/test/scala/org/apache/spark/sql/CollationSQLExpressionsSuite.scala

```diff
@@ -1584,6 +1587,237 @@ class CollationSQLExpressionsSuite
+    })
+  }
+
+  test("TruncDate expression with collation") {
+    // Supported collations
+    testSuppCollations.foreach(collationName => {
+      val query =
+        s"""
+          |select trunc(collate('2016-12-31 23:59:59', '${collationName}'), 'MM')
+          |""".stripMargin
+      // Result & data type check
+      val testQuery = sql(query)
+      val dataType = DateType
+      val expectedResult = "2016-12-01"
+      assert(testQuery.schema.fields.head.dataType.sameType(dataType))
+      checkAnswer(testQuery, Row(Date.valueOf(expectedResult)))
+    })
+  }
+
+  test("TruncTimestamp expression with collation") {
+    // Supported collations
+    testSuppCollations.foreach(collationName => {
+      val query =
+        s"""
+          |select date_trunc(collate('HOUR', '${collationName}'),
+          |collate('2015-03-05T09:32:05.359', '${collationName}'))
+          |""".stripMargin
+      // Result & data type check
+      val testQuery = sql(query)
+      val dataType = TimestampType
+      val expectedResult = "2015-03-05 09:00:00.0"
+      assert(testQuery.schema.fields.head.dataType.sameType(dataType))
+      checkAnswer(testQuery, Row(Timestamp.valueOf(expectedResult)))
+    })
+  }
+
+  test("MakeTimestamp expression with collation") {
+    // Supported collations
+    testSuppCollations.foreach(collationName => {
+      val query =
+        s"""
+          |select make_timestamp(2014, 12, 28, 6, 30, 45.887, collate('CET', '${collationName}'))
+          |""".stripMargin
+      // Result & data type check
+      val testQuery = sql(query)
+      val dataType = TimestampType
+      val expectedResult = "2014-12-27 21:30:45.887"
+      assert(testQuery.schema.fields.head.dataType.sameType(dataType))
+      checkAnswer(testQuery, Row(Timestamp.valueOf(expectedResult)))
+    })
+  }
+
+  test("DatePart expression with collation") {
+    // Supported collations
+    testSuppCollations.foreach(collationName => {
+      val query =
+        s"""
+          |select date_part(collate('Week', '${collationName}'),
+          |collate('2019-08-12 01:00:00.123456', '${collationName}'))
+          |""".stripMargin
+      // Result & data type check
+      val testQuery = sql(query)
+      val dataType = IntegerType
+      val expectedResult = 33
+      assert(testQuery.schema.fields.head.dataType.sameType(dataType))
+      checkAnswer(testQuery, Row(expectedResult))
+    })
+  }
+
+  test("ConvertTimezone expression with collation") {
+    // Supported collations
+    testSuppCollations.foreach(collationName => {
+      val query =
+        s"""
+          |select date_format(convert_timezone(collate('America/Los_Angeles', '${collationName}'),
+          |collate('UTC', '${collationName}'), collate('2021-12-06 00:00:00', '${collationName}')),
+          |'yyyy-MM-dd HH:mm:ss.S')
+          |""".stripMargin
+      // Result & data type check
+      val testQuery = sql(query)
+      val dataType = StringType
```

Comment:
why not StringType(collationName)?

Reply:
date_format here is used just for sake of easier result check, since there is no easy way to check result type of TimestampNZT, therefore I'm formating the output to string. 
Specific test for date_format and collations is added to https://github.com/apache/spark/pull/46561

Reply:
well then no need to test that here if we're already testing it thoroughly in a separate PR?

Reply:
Discussed offline.